### DISCUSSION TITLE
[Feature request] Comfortable export to independent applications.

### DISCUSSION THREAD

**Discussion by GrzechuGrek** [4ian/GDevelop#5579] [Enhancement ideas (and their technical discussions)]
I would like the possibility of easy exports games to standalone applications without struggling with external applications/services and typing console commands. Just like in case of most other game creation programs - directly from menu.

**Comment by sakunix** [comment]
Yes, it is a good idea, especially to export to Android, the command to create the cordova buid does not work for me, I have to use Android Studio to create the manual export.

**Comment by Nintorch** [comment]
I also like this idea because I have also tried other engines where you can either just click from a menu to export (like Clickteam Fusion) or you can ask the engine to download the build tools and files for you before being able to export with one click (Godot)

I have actually started trying implementing this myself but It's not anywhere near perfection yet, as I'm still experimenting with this: https://github.com/Nintorch/GDevelop
Basically, before the GDevelop UI was changed and there was a preferences menu (I haven't found it in the new UI yet), I added a section there to install and setup the build tools (like Node, Electron Builder and other dependencies, this also includes adding it to path environment variable in the gdevelop process), and a new section for game exports, like a "Local export", that uses the build tools installed earlier to build the game.

I'm not sure what approach I should use exactly (like downloading Node and other dependencies when asked; bundle electron builder, if that's even possible; or something else), so I will continue experimenting.

I would really like the GDevelop's team opinion on this topic though.

**Reply by arthuro555** [reply]
I mean the GDevelop Team already provides their take on solving this problem through the one-click export.
Bundling or auto-installing the tools is not very practical, since there are licensing concerns, the installation processes are always platform specific and in the case of some components (like the JDK and Android SDK) not well defined, depending on your platform some build types are not possible, it's hard to stay up-to-date with the build tools while not always using the latest version which might have breaking changes, etc. 

The online build service is the solution to these problems, and takes care of building from the right platform, with the best build tools version, etc. for you. It allows to really have the most comfortable experience possible.

If you want to do a manual build, it is assumed that you know what you're doing and want that additional control over your environment & build tools rather than a one-click solution.

**Reply by Nintorch** [reply]
These are good points, I didn't consider them. Thank you for your response
